### DISCUSSION TITLE
How to run several shell commands in a row?

### DISCUSSION THREAD

**Discussion by alexlabs** [yume-chan/ya-webadb#414] [Q&A] [answered]
How to run several shell commands in a row?  
For example, I need to execute multiple commands consecutively in a function

It should be Like below?
this.adb.subprocess.spawnAndWaitLegacy(['kill', 'system_server']);
this.adb.subprocess.spawnAndWaitLegacy(['stop']);

Thanks

**Comment by yume-chan** [comment, marked as answer]
What result do you want to achieve?

**Reply by alexlabs** [reply]
I have a script that needs to set multiple properties continuously. I want to integrate it into a function. If every command calls the function spawnAndWaitLegacy, will the efficiency be too low?

**Reply by yume-chan** [reply]
You can write inline shell script in `spawnAndWaitLegacy` like `spawnAndWaitLegacy("export a=1; export b=2; echo $a")`

**Reply by alexlabs** [reply]
Thanks a lot! It works well!
